sopel/modules/adminchannel.py

```diff
@@ -112,7 +112,7 @@ def kick(bot, trigger):
         channel = opt
         reasonidx = 3
     reason = ' '.join(text[reasonidx:])
-    if nick != bot.config.core.nick:
+    if nick != bot.make_identifier(bot.config.core.nick):
```

Comment:
Some future patch making `bot.config.core.nick` already return an `Identifier` would probably be very helpful. Should you agree, I look forward to (what I expect will be) a discussion about how annoying that would be to implement. 😛

Reply:
This is simply not possible: the configuration is evaluated at startup, before the bot even exists in the first place. This is the string put in the configuration file, and I'd prefer the configuration to stay that way if possible.

Reply:
Oof, right. Changing the type after bootstrapping (even though `Identifier` is basically `str`) wouldn't be great.

But I still have another question after looking at the context: Why not just use `if nick != bot.nick`?

Reply:
That's actually a good question, why was it like that? I don't know. I don't think I should change that in *that* PR, but it is worth thinking about it.